### ISSUE TITLE
Automatically assign post number to pages

### ISSUE THREAD

**Issue by pagaidu** [11ty/eleventy#1245]
I write to understand how to make Eleventy site smarter so that each post number is defined automatically. So far I assign post number manually like this:
```posts-list.njk``` component:
```
<ol>
{%- for post in collections.postsDescending %}
     <li>
          <a href="{{ post.url | url }}">{{ post.data.title }}</a>
     </li>
{%- endfor -%}
{%- if includeAnnex %}
     <li>
          <a href="/">Annex</a>
     </li>
{%- endif -%}    
</ol>
```

```post.njk``` layout uses ```getNextCollectionItem``` and ```getPreviousCollectionItem``` filters. Current post number assigned thanks to ```getCollectionItem``` filter.
```
{% set previousPost = collections.postsDescending | getPreviousCollectionItem(page) %}
{% set nextPost = collections.postsDescending | getNextCollectionItem(page) %}
{% set currentPost = collections.postsDescending | getCollectionItem(page) %}

<a rel="prev" class="link-sm" href="{{ previousPost.url | url }}">Previous</a>
<a rel="next" class="link-sm" href="{{ nextPost.url | url }}">Next</a>

<article class="post">
     {{ content | safe }}
     {{ currentPost.data.number }}
</article> 
```
``` currentPost.data.number ``` is defined in Front Matter manually in every post markdown file, f.ex. ```post-1.md```:
```
---
title: lorem ipsum
tags:
  - another-tag
layout: layouts/post.njk
number: 1
---
lorem ipsum
```

My CSS for ordered list numbers (dont know if is relevant):
```
ol li {
    counter-increment: list;
    position: relative;
}
ol li:before {
    content: counter(list) "";
    left: -2.5em;
    position: absolute;
    text-align: right;
    width: 2em;    
}
```
Hope someone will help to figure out this one.

**Comment by pagaidu** [comment]
Eleventy community reminds of https://www.youtube.com/watch?v=dSINO6MKtco GRAMMOPHONE - Rowan Atkinson - Not The Nine o'Clock News
No clues ...

**Comment by zachleat** [comment]
Yes hello hi, I am here 5 years later. If you’re using `<ol>` you can rely on the counter there. We use a pattern on eleventy-base-blog to change the start index (and reverse the numbers provided). https://github.com/11ty/eleventy-base-blog/blob/e775f83c2bac667ae9989a8e3a90850d41b89d82/_includes/postslist.njk#L1

You can even change the numbering pattern using CSS

https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/ol and https://developer.mozilla.org/en-US/docs/Web/CSS/Reference/Properties/list-style-type

Alternatively, you can use `loop.index` or `loop.index0` provided by Nunjucks when in a `{% for %}`: https://mozilla.github.io/nunjucks/templating.html#for
